core/src/git.rs

```diff
@@ -0,0 +1,51 @@
+use std::fs;
+
+use git2::Repository;
+
+use crate::basic_config::BasicConfigContent;
+
+pub struct GitModule {}
+
+impl GitModule {
+    pub fn git_clone(source_path: &str, target_folder: &str) -> anyhow::Result<()> {
+        if let Some((parent, _)) = target_folder.rsplit_once("/") {
+            fs::create_dir_all(parent)?;
+        };
+
+        let repo = match Repository::clone(source_path, target_folder) {
+            Ok(repository) => repository,
+            Err(err) => {
+                println!("Git clone failed with following error: {}", err.message());
+                return Ok(());
+            }
+        };
+
+        println!("cloned repository: {:?}", repo.workdir());
+
+        Ok(())
+    }
```

Comment:
This whole block feels awkward. I wouldn't normally match on a result type in the first place but it looks like you're trying to just gracefully handle the failure and succeed which you probably should do more like:

```suggestion
        Repository::clone(source_path, target_folder)
            .map(|repo| {
                println!("cloned repository: {:?}", repo.workdir());
            })
            .unwrap_or_else(|err| {
                println!("Git clone failed with following error: {}", err.message());
            });

        Ok(())
    }
```